proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -515,4 +523,46 @@ private boolean isIncognitoEnabled(@Nonnull final Map<String, Object> inputConte
                 .map(value -> value.equalsIgnoreCase("true"))
                 .orElse(false);
     }
+
+    private void populateProperties(
+            final String testName,
+            final TestBucket testBucket,
+            final Map<String, PayloadProperty> testProperties) {
+        final Payload payload = testBucket.getPayload();
+        if (payload != null && payload.getJson() != null) {
+            payload.getJson()
+                    .fields()
+                    .forEachRemaining(
+                            field -> {
```

Comment:
Two things:

1. I think I recall a conversation between you and Steven about the code duplication and why its required. Are the reasons documented somewhere? I think it would be helpful to reference
2. nit: I think this should be a separate method

Reply:
> I think I recall a conversation between you and Steven about the code duplication and why its required. Are the reasons documented somewhere? I think it would be helpful to reference

The code duplication should be lessened now that some PRs have been merged to the feature branch. The issue is that the branches on my fork can not be created as PRs being merged into each other in this project, they need to merge into a branch that exists within the original repo not my fork. 

> nit: I think this should be a separate method

I will move this to a function. thanks!